src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -454,6 +461,53 @@ public AgentSettings LoadSettings()
             }
         }
 
+#if OS_WINDOWS
+        private void ConfigureAutoLogonIfNeeded(CommandSettings command)
+        {
+            Trace.Entering();
+
+            if (!command.GetEnableAutoLogon())
+            {
+                return;
+            }
+
+            Trace.Info("Agent is going to run as process setting up the 'AutoLogon' capability for the agent.");
+            try
+            {                
+                var autoLogonConfigManager = HostContext.GetService<IAutoLogonConfigurationManager>();
+                autoLogonConfigManager.Configure(command);
+
+                if (autoLogonConfigManager.RestartNeeded())
```

Comment:
The agent process doesnt have any dependency on the settings that got changed.  It is the env that  reads about these settings (e.g., screen saver) so we should not restart in case of same user.

---
In reply to: [118143584](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118143584) [](ancestors = 118143584)